message/query_options.go

```diff
@@ -322,3 +339,31 @@ func DecodeQueryOptions(source io.Reader, version primitive.ProtocolVersion) (op
 	}
 	return options, nil
 }
+
+func CloneValuesSlice(o []*primitive.Value) []*primitive.Value {
+	var newValues []*primitive.Value
+	if o != nil {
+		newValues = make([]*primitive.Value, len(o))
+		for idx, v := range o {
+			newValues[idx] = v.Clone()
+		}
+	} else {
+		newValues = nil
+	}
+
+	return newValues
+}
+
+func CloneValuesMap(o map[string]*primitive.Value) map[string]*primitive.Value {
```

Comment:
Same => `NamedValues` type?

Reply:
Discussed offline, this involves changing the existing code that encodes and decodes these protocol types so it's out of scope of this PR.